### PR TITLE
Add ResNeSt-26 and Fix Horovod ImageNet training

### PR DESCRIPTION
**Pull request by Jerryzcn**
fix imagenet horovod training bug
fix rcnn custom model syncbn bug
add resnest26
fix resnest

### CODE REVIEW

File: gluoncv/model_zoo/model_zoo.py [hunk 1]
```diff
@@ -54,6 +54,7 @@
     'resnet50_v2': resnet50_v2,
     'resnet101_v2': resnet101_v2,
     'resnet152_v2': resnet152_v2,
+    'resnest26': resnest26,
```
Comment (hetong007) [inline gluoncv/model_zoo/model_zoo.py:57]: Could you please share the context for adding this model?

**Comment by mli** [comment]
Job PR-1263-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1263/2/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/model_zoo/model_zoo.py (hunk 1 above)
Comment (Jerryzcn) [reply gluoncv/model_zoo/model_zoo.py:57]: this is a lightweight model for resnest.

**Comment by mli** [comment]
Job PR-1263-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1263/5/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1263-6 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1263/6/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1263-8 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1263/8/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**